File: segundos 4 lenguajes/Java/JugadorDefensa.java
package Java;
abstract class JugadorDefensa {
    String nombre;
    int nivel;

    public JugadorDefensa() {}

    public JugadorDefensa(String nombre) {
        this.nombre = nombre;
    }

    public JugadorDefensa(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorDefensa(JugadorDefensa j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }

    abstract void defender();
}

File: segundos 4 lenguajes/go/JugadorDefensa.go
package main

type JugadorDefensa struct {
	Nombre string
	Nivel  int
}

func NewJugadorDefensa() JugadorDefensa {
	return JugadorDefensa{"", 0}
}

func NewJugadorDefensaNombre(nombre string) JugadorDefensa {
	return JugadorDefensa{nombre, 1}
}

func NewJugadorDefensaParams(nombre string, nivel int) JugadorDefensa {
	return JugadorDefensa{nombre, nivel}
}

func NewJugadorDefensaCopy(j JugadorDefensa) JugadorDefensa {
	return JugadorDefensa{j.Nombre, j.Nivel}
}

type AbstractDefensa interface {
	Defender()
}
